docs/content/sips/011-component-versioning.md

```diff
@@ -0,0 +1,41 @@
+title = "SIP 011 - Component versioning"
+template = "main"
+date = "2023-01-24T01:01:01Z"
+---
+
+Summary: Embed version information into Spin components for improved compatibility diagnostics.
+
+Owner(s): joel.dice@fermyon.com
+
+Created: Jan 24, 2023
+
+## Background
+
+Currently, Spin has no way to determine which version of SDK the components of an application were built with, which leads to obscure, hard-to-understand error messages for ABI and/or API incompatibilities.  For example, a component built with a Spin 0.4 SDK won't run on Spin 0.7 due to `wit-bindgen` ABI changes, leading to a variety of confusing runtime errors like out-of-bounds memory traps, assertion errors about invalid values, etc.  In addition, new APIs were added between 0.4 and 0.7, and thus a 0.7 component might not run on 0.4 even if the ABI had not changed.  These errors are usually a bit easier to understand (e.g. missing imports), but still don't point clearly to a resolution.  Ideally, the message presented to the user would be something like, "This component was built with a Spin 0.7 SDK and cannot be run using this version of Spin (0.4) -- please use Spin 0.7 or later."
+
+Note that when we say "SDK" in this document, we mean either one of the official Spin SDKs _or_ raw bindings generated from the `WIT` files and the `wit-bindgen` version shipped with a given version of Spin, e.g. for languages which do not yet have an "official" SDK.
+
+## Proposal
+
+Although `WIT` does not yet have a way to express API versions, we can emulate it by introducing a new interface called `spin-version.wit` containing a single function to be exported by the guest component:
```

Comment:
Maybe `spin-sdk-version`? `spin-version` sounds like the reverse; giving the host version to the guest.